### ISSUE TITLE
Can't Sign in to Sync

### ISSUE THREAD

**Issue by iShyboy** [MrAlex94/Waterfox#2718]
### What happened?

Can't sync.
I can login in accounts.firefox.com but when I tap Sign in to Sync it gives me my account but ask for the pass, after typing it I receive an endless load and a 'working' message.

![image](https://user-images.githubusercontent.com/10236047/173976335-fa240b7f-f6be-43e8-a5ee-4069ab859ea0.png)

I've used a few addons vpn and a local machine vpn to test if I could sync too, not helping.

### Reproducible?

- [ ] I have tested this on Firefox.

### Version

G4

### What platform are you seeing the problem on?

Windows

**Comment by VCoklat** [comment]
Yep I got the same problem will try to downgrade the version first

**Comment by Shi48C** [comment]
I have the same issue also.

I've tried it in Windows 10 & 11, on 2 different PCs, tried the latest version and an older one I had on another computer & I've tried 2 different internet connections.

It works fine in Firefox but I don't really want to use Firefox if I can help it.

I get the following in the console:
_Content Security Policy: Couldn’t process unknown directive ‘script-src-attr’
Response not received for: fxaccounts:can_link_account_

**Comment by NoelleTGS** [comment]
I'm getting the same on Linux but I can't even get past the email screen.
Console output:
```
Working… logger.js:69:29
Response not received for: fxaccounts:can_link_account logger.js:72:35
```

**Comment by tonyclemmey** [comment]
Experiencing the same issue on MacOS Big Sur, Waterfox G4.1.2.1

**Comment by JackASil** [comment]
Same here. G4.1.3

**Comment by accordeonixte** [comment]
Same for G4.1.3.1

**Comment by iShyboy** [comment]
> Yep I got the same problem will try to downgrade the version first

Didn't need to go that far, just copy pasted profile from current firefox already synced.

**Comment by sengluty198** [comment]
same

**Comment by kuator** [comment]
Same problem

**Comment by daveedmee** [comment]
> > Yep I got the same problem will try to downgrade the version first
> 
> Didn't need to go that far, just copy pasted profile from current firefox already synced.

How did you do that? I've tried and Waterfox was unable to start after that, got error message that my profiles couldn't be loaded

**Comment by Rokta** [comment]
I had same problem and resolved it by following these steps: 
https://www.reddit.com/r/firefox/comments/5d7cid/cannot_log_in_to_firefox_sync_stuck_on_working/

Steps:
open about:config
add "https://accounts.firefox.com/" to "webchannel.allowObject.urlWhitelist" config option
setup sync

**Comment by JackASil** [comment]
> I had same problem and resolved it by following these steps: https://www.reddit.com/r/firefox/comments/5d7cid/cannot_log_in_to_firefox_sync_stuck_on_working/
> 
> Steps: open about:config add "https://accounts.firefox.com/" to "webchannel.allowObject.urlWhitelist" config option setup sync

That does solve the issue. Thanks!

**Comment by MememeOyaji** [comment]
I could NOT sign in although I had done the steps which Rokta showed.
To downgrade to G4.0.1 was necessary. (without changing about:config)

**Comment by kamil445** [comment]
I had this issue too...

> I had same problem and resolved it by following these steps: https://www.reddit.com/r/firefox/comments/5d7cid/cannot_log_in_to_firefox_sync_stuck_on_working/
> 
> Steps: open about:config add "https://accounts.firefox.com/" to "webchannel.allowObject.urlWhitelist" config option setup sync

worked for me

**Comment by iShyboy** [comment]
> I had same problem and resolved it by following these steps: https://www.reddit.com/r/firefox/comments/5d7cid/cannot_log_in_to_firefox_sync_stuck_on_working/
> 
> Steps: open about:config add "https://accounts.firefox.com/" to "webchannel.allowObject.urlWhitelist" config option setup sync

This did the job.

**Comment by adamp01** [comment]
Apologies for getting to this so late, am looking into what is actually causing this.

**Comment by adamp01** [comment]
The root cause for this is related to user agent. Am going to push a fix for it shortly.

**Comment by daveedmee** [comment]
> The root cause for this is related to user agent. Am going to push a fix for it shortly.

So does that mean it could be solved with add-ons like an user agent switcher?

**Comment by daveedmee** [comment]
> I had same problem and resolved it by following these steps: https://www.reddit.com/r/firefox/comments/5d7cid/cannot_log_in_to_firefox_sync_stuck_on_working/
> 
> Steps: open about:config add "https://accounts.firefox.com/" to "webchannel.allowObject.urlWhitelist" config option setup sync

It's also possible to install Firefox, sign in, import data from Firefox to Waterfox and uninstall Firefox after that. You'll automatically be logged in after a restart of Waterfox.

**Comment by adamp01** [comment]
> > The root cause for this is related to user agent. Am going to push a fix for it shortly.
> 
> So does that mean it could be solved with add-ons like an user agent switcher?

Not sure if a UA switcher will actually work, but overriding the UA for the browser by doing the below will work:
- Go to `about:config`
- Add a new string preference called `general.useragent.override`.
- Set the value to a relevant value from: https://www.whatismybrowser.com/guides/the-latest-user-agent/firefox

It's another way of achieving the same result that has already been mentioned.

**Comment by MememeOyaji** [comment]
New version G4.1.3.2 can sign in to sync without change about:config.
